src/System.Runtime.Numerics/src/System/Numerics/BigInteger.cs

```diff
@@ -552,17 +552,8 @@ public BigInteger(ulong value)
         }
 
         public BigInteger(Single value)
+            : this((Double)value)
         {
-            if (Single.IsInfinity(value))
-                throw new OverflowException(SR.Overflow_BigIntInfinity);
-            if (Single.IsNaN(value))
-                throw new OverflowException(SR.Overflow_NotANumber);
```

Comment:
@CarolEidt, could you clarify for us some foating-point-related behavior?  Are the following two examples semantically equivalent with regards to infinity/NaN checking:
```C#
public BigInteger(Single value) : this((Double)value)
{
}

public BigInteger(Double value)
{
    if (Double.IsInfinity(value)) throw ...;
    if (Double.IsNaN(value)) throw ...;
    ...;
}
```
and
```C#
public BigInteger(Single value)
{
    if (Single.IsInfinity(value)) throw ...;
    if (Single.IsNaN(value)) throw ...;
    ...
}

public BigInteger(Double value)
{
    if (Double.IsInfinity(value)) throw ...;
    if (Double.IsNaN(value)) throw ...;
    ...
}
```
?